packages/core/src/withMemoryCache.ts

```diff
@@ -0,0 +1,24 @@
+import { SearchFn, SearchOptions } from "@core/types"
+import { SearchQuery, SearchResult } from "@nosto/nosto-js/client"
+
+import { getFromCache, setCache } from "./utils/memoryCache"
+
+export async function searchWithMemoryCache(
+  query: SearchQuery,
+  { track, ...options }: SearchOptions,
+  searchFn: SearchFn
+): Promise<SearchResult> {
+  // For autocomplete only
+  if (track !== "autocomplete") {
```

Comment:
make this decision at a higher level
this middleware should get activated based on a flag in search options